server/core/src/main/java/dev/slimevr/tracking/processor/skeleton/LegTweaks.kt

```diff
@@ -0,0 +1,1167 @@
+		private const val WAIST_PUSH_WEIGHT = 0.2f
+
+		// COM calculation
+		private const val HEAD_MASS = 0.0827f
+		private const val THORAX_MASS = 0.1870f
+		private const val ABDOMEN_MASS = 0.1320f
+		private const val PELVIS_MASS = 0.1530f
+		private const val THIGH_MASS = 0.1122f
+		private const val LEG_AND_FOOT_MASS = 0.0620f
+		private const val UPPER_ARM_MASS = 0.0263f
+		private const val FOREARM_AND_HAND_MASS = 0.0224f
+
+		// rotation correction
+		private const val ROTATION_CORRECTION_VERTICAL = 0.1f
+		private const val MAXIMUM_CORRECTION_ANGLE = 0.4f
+		private const val MAXIMUM_CORRECTION_ANGLE_DELTA = 0.7f
+		private const val MAXIMUM_TOE_DOWN_ANGLE = 0.8f
+		private const val TOE_SNAP_COOLDOWN = 3.0f
+		private const val MIN_DISTANCE_FOR_PLANT = 0.025f
+		private const val MAX_DISTANCE_FOR_PLANT = 0.075f
+
+		// misc
+		const val NEARLY_ZERO = 0.001f
+		private const val STANDING_CUTOFF_VERTICAL = 0.65f
+		private const val MAX_DISENGAGEMENT_OFFSET = 0.30f
+		private const val DEFAULT_ARM_DISTANCE = 0.15f
+		private const val MAX_CORRECTION_STRENGTH_DELTA = 1.0f
+	}
+
+	// state variables
+	private var floorLevel = 0f
+	private var hipToFloorDist = 0f
+	private var currentDisengagementOffset = 0.0f
+	private var footLength = 0.0f
+	private var currentCorrectionStrength = 0.3f // default value
+
+	private var initialized = true
+	private var enabled = true // master switch
+
+	private var floorClipEnabled = false
+	private var alwaysUseFloorclip = false
+	private var skatingCorrectionEnabled = false
+	private var toeSnapEnabled = false
+	private var footPlantEnabled = false
+	private var active = false
+	private var rightLegActive = false
+	private var leftLegActive = false
+	private var leftFramesLocked = 0
+	private var rightFramesLocked = 0
+	private var leftFramesUnlocked = 0
+	private var rightFramesUnlocked = 0
+	private var leftToeAngle = 0.0f
+	private var leftToeTouched = false
+	private var rightToeAngle = 0.0f
+	private var rightToeTouched = false
+	private var localizerMode = false
+
+	// config
+	private var config: LegTweaksConfig? = null
+
+	// leg data
+	private var leftFootPosition = Vector3.NULL
+	private var rightFootPosition = Vector3.NULL
+	private var leftKneePosition = Vector3.NULL
+	private var rightKneePosition = Vector3.NULL
+	private var hipPosition = Vector3.NULL
+	private var leftFootRotation = Quaternion.IDENTITY
+	private var rightFootRotation = Quaternion.IDENTITY
+
+	private var leftFootAcceleration = Vector3.NULL
+	private var rightFootAcceleration = Vector3.NULL
+	private var leftLowerLegAcceleration = Vector3.NULL
+	private var rightLowerLegAcceleration = Vector3.NULL
+
+	// buffer for holding previous frames of data
+	private var bufferHead = LegTweaksBuffer()
+	private var bufferInvalid = true
+
+	constructor(skeleton: HumanSkeleton, config: LegTweaksConfig) : this(skeleton) {
+		this.config = config
+		updateConfig()
+	}
+
+	fun getLeftFootPosition(): Vector3 {
+		return leftFootPosition
+	}
```

Comment:
I would just make all of these public variables and remove the get/set

Reply:
didn't know Kotlin had that feature! thanks